src/test/lib/optimizer/strategy/predicate_placement_rule_test.cpp

```diff
@@ -584,6 +585,50 @@ TEST_F(PredicatePlacementRuleTest, SemiPushDown) {
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
 }
 
+TEST_F(PredicatePlacementRuleTest, HandleBarrierPredicatePushdown) {
+  // In this test, the PredicatePlacementRule cannot push down the top two predicates because of a predicate having
+  // multiple outputs (the barrier node). The purpose of this test is to check whether a barrier node becomes
+  // pushed down by the rule, if it is a predicate eligible for pushdown.
+
+  // clang-format off
+  const auto barrier_predicate_node =
+  PredicateNode::make(greater_than_(_b_b, 123),                 // <-- 1st Predicate before pushdown (pushdown barrier due to output_count() == 2) // NOLINT
+    JoinNode::make(JoinMode::Semi, equals_(_a_a, _b_a),         // <-- 2nd Predicate before pushdown
```

Comment:
Regarding `2nd Predicate`: If someone missed the following line in `predicate_placement_rule.cpp`, calling this join node a Predicate might be confusing.

>// We pick up semi and anti joins on the way and treat them as if they were predicates

How about `Join node treated as 2nd Predicate before pushdown`?

Reply:
The test above is called "SemiPushDown". Also, the `PredicatePlacementRule` states in the header Doc:

> For the purpose of this rule, semi- and anti-joins are treated as predicates as well.


Therefore, I think this is redundant. One should familiarize with the rule itself before fiddling around with its tests.